test/fluree/db/reasoner/owl_datalog_test.clj

```diff
@@ -0,0 +1,401 @@
+
+(deftest ^:integration equivalentClass-with-intersection-test
+  (testing "owl:equivalentClass with intersection of named class and restrictions"
+    ;; This pattern is used extensively in gistPharma
+    ;; Example: gist:DrugProduct ≡ (ManufacturedItem ∩ SubstanceDefinition ∩ ∃conformsTo.Formulation)
+    (let [conn (test-utils/create-conn)
+          db @(fluree/create conn "reasoner/equiv-intersection" nil)
+
+          ;; Define the ontology with complex equivalentClass
+          ontology {"@context" {"ex"   "http://example.org/"
+                                "owl"  "http://www.w3.org/2002/07/owl#"
+                                "rdfs" "http://www.w3.org/2000/01/rdf-schema#"}
+                    "insert"   [{"@id"                  "ex:DrugProduct"
+                                 "@type"                "owl:Class"
+                                 "owl:equivalentClass"  {"@type"              "owl:Class"
+                                                         "owl:intersectionOf" [{"@id" "ex:ManufacturedItem"}
+                                                                               {"@id" "ex:SubstanceDefinition"}
+                                                                               {"@type"              "owl:Restriction"
+                                                                                "owl:onProperty"     {"@id" "ex:conformsTo"}
+                                                                                "owl:someValuesFrom" {"@id" "ex:Formulation"}}]}}
+                                {"@id"   "ex:ManufacturedItem"
+                                 "@type" "owl:Class"}
+                                {"@id"   "ex:SubstanceDefinition"
+                                 "@type" "owl:Class"}
+                                {"@id"   "ex:Formulation"
+                                 "@type" "owl:Class"}
+                                {"@id"   "ex:conformsTo"
+                                 "@type" "owl:ObjectProperty"}]}
+
+          db-with-ontology @(fluree/update db ontology)
+
+          ;; Add instance data that should be inferred as DrugProduct
+          instance-data {"@context" {"ex" "http://example.org/"}
+                         "insert"   {"@id"          "ex:aspirin-tablet"
+                                     "@type"        ["ex:ManufacturedItem" "ex:SubstanceDefinition"]
+                                     "ex:conformsTo" {"@id"   "ex:aspirin-formulation"
+                                                      "@type" "ex:Formulation"}}}
+
+          db-with-data @(fluree/update db-with-ontology instance-data)
+
+          ;; Apply owl-datalog reasoning
+          db-reasoned @(fluree/reason db-with-data :owl-datalog)]
+
+      (testing "Instance with all intersection components should be inferred as DrugProduct"
+        (is (= #{"ex:DrugProduct" "ex:ManufacturedItem" "ex:SubstanceDefinition"}
+               (set @(fluree/query db-reasoned
```

Comment:
Do we need to coerce the results into a set? By my read the output should be deterministic.